### PR TITLE
feat: Migrates some resources to new file structure

### PR DESCRIPTION
**Pull request by lantoli**
## Description

Migrates some resources to new file structure:

- cloud_backup_schedule [CLOUDP-214973](https://jira.mongodb.org/browse/CLOUDP-214973)
- cloud_backup_snapshot [CLOUDP-215020](https://jira.mongodb.org/browse/CLOUDP-215020)
- cloud_backup_snapshot_export_bucket [CLOUDP-215016](https://jira.mongodb.org/browse/CLOUDP-215016)
- cloud_backup_snapshot_export_job [CLOUDP-215000](https://jira.mongodb.org/browse/CLOUDP-215000)
- cloud_backup_snapshot_restore_job [CLOUDP-214967](https://jira.mongodb.org/browse/CLOUDP-214967)
- backup_compliance_policy [CLOUDP-214991](https://jira.mongodb.org/browse/CLOUDP-214991)

## Type of change:

- [ ] Bug fix (non-breaking change which fixes an issue). Please, add the "bug" label to the PR.
- [ ] New feature (non-breaking change which adds functionality). Please, add the "enhancement" label to the PR.
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected). Please, add the "breaking change" label to the PR.
- [ ] This change requires a documentation update
- [ ] Documentation fix/enhancement

## Required Checklist:

- [x] I have signed the [MongoDB CLA](https://www.mongodb.com/legal/contributor-agreement)
- [x] I have read the [contribution guidelines](https://github.com/mongodb/terraform-provider-mongodbatlas/blob/master/CONTRIBUTING.md)
- [x] I have checked that this change does not generate any credentials and that **they are NOT accidentally logged anywhere**.
- [x] I have added tests that prove my fix is effective or that my feature works per HashiCorp requirements
- [x] I have added any necessary documentation (if appropriate)
- [x] I have run make fmt and formatted my code
- [x] If changes include deprecations or removals, I defined an isolated PR with a relevant title as it will be used in the auto-generated changelog.

### CODE REVIEW

File: internal/service/advancedcluster/data_source_advanced_clusters.go [hunk 1]
```diff
@@ -14,7 +14,7 @@ import (
 	matlas "go.mongodb.org/atlas/mongodbatlas"
 )
 
-func DataSourceAdvancedClusters() *schema.Resource {
+func PluralDataSource() *schema.Resource {
```
Comment (andreaangiolillo) [inline internal/service/advancedcluster/data_source_advanced_clusters.go:17]: (not blocking) Interesting name 🤔 What do you think about `DataSources` instead? 
```suggestion
func DataSources() *schema.Resource {
```

File: internal/provider/provider_sdk2.go [hunk 2]
```diff
@@ -186,12 +192,12 @@ func getDataSourcesMap() map[string]*schema.Resource {
 
 func getResourcesMap() map[string]*schema.Resource {
 	resourcesMap := map[string]*schema.Resource{
-		"mongodbatlas_advanced_cluster":                  advancedcluster.ResourceAdvancedCluster(),
+		"mongodbatlas_advanced_cluster":                  advancedcluster.Resource(),
```
Comment (andreaangiolillo) [inline internal/provider/provider_sdk2.go:195]: Do you need to update the github actions with the new path?

**Review by andreaangiolillo — APPROVED** [review]
LGTM 💯

**Review by AgustinBettati — APPROVED** [review]
LGTM

File: internal/provider/provider_sdk2.go (hunk 2 above)
Comment (lantoli) [reply internal/provider/provider_sdk2.go:195]: oops, you're right!

File: internal/service/advancedcluster/data_source_advanced_clusters.go (hunk 1 above)
Comment (lantoli) [reply internal/service/advancedcluster/data_source_advanced_clusters.go:17]: I'm following what @AgustinBettati started, not a strong opinion to either

File: .github/workflows/test-suite.yml [hunk 3]
```diff
@@ -3,7 +3,7 @@ name: 'Test Suite'
 on:
   workflow_dispatch:
   schedule:
-    - cron: "0 2 * * *" # workflow runs every day at 02:00 AM
+    - cron: "0 0 * * *" # workflow runs every day at midnight UTC
```
Comment (lantoli) [inline .github/workflows/test-suite.yml:6]: to hav more time

File: .github/workflows/test-suite.yml [hunk 4]
```diff
@@ -18,7 +18,7 @@ jobs:
       max-parallel: 1
       fail-fast: false
       matrix:
-        terraform_version: ['', '1.1.6', '1.0.8'] # '' for latest version
+        terraform_version: ['', '1.0.8'] # '' for latest version
```
Comment (lantoli) [inline .github/workflows/test-suite.yml:21]: we already have the data for 1.1.6, removed to decrease time

**Pull request merged**